### PR TITLE
User profile picture upload feature

### PR DESCRIPTION
**Pull request by Zekereyez**
In this PR I implemented a feature for the user to upload their profile pictures and allow the user to scroll through them horizontally. I did however edit the launch icons due to them looking pretty bad when running the app on iphone (sorry). There may be a bug with this feature, when uploading a new image, a previous image loads first before the new image takes its place.  

Demo here: https://imgur.com/a/3cy19N9

Also... the creation of user profile pictures implies the deletion of them, right?...

### CODE REVIEW

File: Squablr/Models/Profile.h [hunk 1]
```diff
@@ -18,8 +18,9 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, strong) NSString *stance;
 @property (nonatomic, strong) NSNumber *experience;
 @property (nonatomic, strong) NSString *biography;
-@property (nonatomic, strong) PFFileObject *profileImages;
+@property (nonatomic, strong) PFFileObject *imageFile;
 +(void) writeUserToParse: (PFBooleanResultBlock  _Nullable)completion;
```
Comment (rhammoud0418) [inline Squablr/Models/Profile.h]: I would put an extra space here in between + and (

File: Squablr/View Controllers/LoginViewController.m [hunk 2]
```diff
@@ -122,7 +122,8 @@ - (IBAction)signIn:(id)sender {
                         [self navigateToUserFeed];
                     }];
                 }
-            } else {
```
Comment (rhammoud0418) [inline Squablr/View Controllers/LoginViewController.m:125]: Why did you change the formatting here? It was correct to begin with.

File: Squablr/View Controllers/ProfilePictureCell.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfilePictureCell.h:1]: This class is a View object, it belongs in the 'Views' folder

File: Squablr/View Controllers/ProfileViewController.h [hunk 4]
```diff
@@ -6,8 +6,12 @@
 //
 
 #import <UIKit/UIKit.h>
-#import "Parse/Parse.h"
+#import <Parse/Parse.h>
+//#import "ParseUI/ParseUI.h"
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.h]: Commented out code

File: Squablr/View Controllers/ProfileViewController.h [hunk 5]
```diff
@@ -21,6 +25,10 @@ NS_ASSUME_NONNULL_BEGIN
 @property (weak, nonatomic) IBOutlet UILabel *userExperience;
 @property (weak, nonatomic) IBOutlet UILabel *userBio;
 @property (weak, nonatomic) IBOutlet UILabel *userProfileName;
+//@property (weak, nonatomic) IBOutlet UIImageView *userProfilePic;
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.h]: Commented out code

File: Squablr/View Controllers/ProfileViewController.m [hunk 6]
```diff
@@ -60,21 +138,58 @@ -(void) queryUserProfileInfo {
...
         self.userBio.text = [NSString stringWithFormat:@"%@", bio];
     }
 }
 
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    // placeholder info for the time being
+    ProfilePictureCell *cell = [self.gridView dequeueReusableCellWithReuseIdentifier:@"ProfilePictureCell" forIndexPath:indexPath];
+    NSLog(@"%@",self.userProfileInfo);
+    cell.profileImage.file = self.userProfilePhotos[indexPath.item];
+//    NSLog(@"%@", self.userProfilePhotos);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Commented out code

File: Squablr/View Controllers/ProfilePictureCell.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+//  Squablr
+//
+//  Created by Zeke Reyes on 7/14/22.
+//
+
+#import <UIKit/UIKit.h>
+#import "Parse/PFImageView.h"
+NS_ASSUME_NONNULL_BEGIN
+
+@interface ProfilePictureCell : UICollectionViewCell
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfilePictureCell.h:12]: I would put a space after this line

File: Squablr/View Controllers/ProfilePictureCell.h [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+@property (weak, nonatomic) IBOutlet PFImageView *profileImage;
+
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfilePictureCell.h:14]: And then remove this space (this is common convention)

File: Squablr/View Controllers/ProfileViewController.m [hunk 9]
```diff
@@ -20,11 +19,90 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
     [self queryUserProfileInfo];
+    self.gridView.dataSource = self;
+    self.gridView.delegate = self;
+    UICollectionViewFlowLayout *layout = (UICollectionViewFlowLayout *) self.gridView.collectionViewLayout;
+    layout.minimumLineSpacing = 0;
+    layout.minimumInteritemSpacing = 0;
+    NSLog(@"%f", self.gridView.frame.size.width);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m:27]: Delete this log statement

File: Squablr/View Controllers/ProfileViewController.m [hunk 10]
```diff
@@ -20,11 +19,90 @@ - (void)viewDidLoad {
...
+    }
+    else {
+        self.userProfilePhotos = [[NSMutableArray alloc] initWithObjects:imageFile, nil];
+    }
+    // Update the Parse image array with the local array of photos
+    user[@"profile"][@"profileImages"] = self.userProfilePhotos;
+    [user saveInBackground];
+    [self.gridView reloadData];
+    
+    NSLog(@"%@", self.userProfilePhotos[0]);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Get rid of this log statement

File: Squablr/View Controllers/ProfileViewController.m [hunk 11]
```diff
@@ -60,21 +137,53 @@ -(void) queryUserProfileInfo {
...
 
+- (void)queryUserImages {
+    PFQuery *profileQuery = [PFQuery queryWithClassName:@"Profile"];
+    [profileQuery whereKey:@"name" equalTo:[PFUser currentUser].username];
+    // fetch data asynchronously
+    [profileQuery findObjectsInBackgroundWithBlock:^(NSArray *profiles, NSError * _Nullable error) {
+        if (profiles) {
+            // Handle fetched data
+            [self.gridView reloadData];
+            NSLog(@"%@", self.userProfilePhotos);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Get rid of this log statement

File: Squablr/View Controllers/ProfileViewController.m [hunk 12]
```diff
@@ -60,21 +137,53 @@ -(void) queryUserProfileInfo {
...
+        }
+        else {
+            // Log error
+            NSLog(@"%@", error.localizedDescription);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Get rid of this log statement

File: Squablr/View Controllers/ProfileViewController.m [hunk 13]
```diff
@@ -60,21 +137,53 @@ -(void) queryUserProfileInfo {
     }];
 }
 
+- (void)queryUserImages {
+    PFQuery *profileQuery = [PFQuery queryWithClassName:@"Profile"];
+    [profileQuery whereKey:@"name" equalTo:[PFUser currentUser].username];
+    // fetch data asynchronously
+    [profileQuery findObjectsInBackgroundWithBlock:^(NSArray *profiles, NSError * _Nullable error) {
+        if (profiles) {
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Why are we querying here? Seems like you aren't really doing anything with the profiles that are returned

File: Squablr/View Controllers/ProfileViewController.m [hunk 14]
```diff
@@ -60,21 +137,53 @@ -(void) queryUserProfileInfo {
...
         self.userExperience.text = [NSString stringWithFormat:@"%@", experience];
-        NSString *bio = [self.arrayOfUserInfo[0] biography];
+        NSString *bio = [self.userProfileInfo[0] biography];
         self.userBio.text = [NSString stringWithFormat:@"%@", bio];
     }
 }
 
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    ProfilePictureCell *cell = [self.gridView dequeueReusableCellWithReuseIdentifier:@"ProfilePictureCell" forIndexPath:indexPath];
+    NSLog(@"%@",self.userProfileInfo);
```
Comment (rhammoud0418) [inline Squablr/View Controllers/ProfileViewController.m]: Get rid of this log statement

**Review by rhammoud0418 — COMMENTED** [review]
Awesome that you were able to get this working! Noticed a few issues so left some feedback.

File: Squablr/View Controllers/LoginViewController.m (hunk 2 above)
Comment (Zekereyez) [reply Squablr/View Controllers/LoginViewController.m:125]: That's weird... it shows up correctly in my file

File: Squablr/Models/Profile.h (hunk 1 above)
Comment (Zekereyez) [reply Squablr/Models/Profile.h]: Got it

File: Squablr/View Controllers/ProfileViewController.m (hunk 13 above)
Comment (Zekereyez) [reply Squablr/View Controllers/ProfileViewController.m]: That's true, will fix this.
